content/docs/user-guide/data-management/cloud-versioning.md

```diff
@@ -0,0 +1,74 @@
+# Cloud Versioning
+
+`dvc remote` storage normally uses
+[content-addressible storage](/doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory)
+to organize versioned data. Different versions of files are stored in the remote
+according to hash of their data content instead of according to their original
+filenames and directory location. This allows DVC to optimize certain remote
+storage lookup and data sync operations, and provides data de-duplication at the
+file level. However, this comes with the drawback of losing human-readable
+filenames without the use of the DVC CLI (`dvc get --show-url`) or API
+(`dvc.api.get_url()`).
+
+DVC supports the use of cloud object versioning for cases where users prefer to
+retain their original filenames and directory hierarchy in remote storage, in
+exchange for losing the de-duplication and performance benefits of
+content-addressible storage. When cloud versioning is enabled, DVC will store
+files in the remote according to their original directory location and
+filenames. Different versions of a file will then be stored as separate versions
+of the corresponding object in cloud storage.
```

Comment:
> When cloud versioning is enabled, DVC will store
files in the remote according to their original directory location and
filenames. Different versions of a file will then be stored as separate versions
of the corresponding object in cloud storage.

I'd suggest starting Cloud Versioning with this part, and then explain the differences from the default content-addressable storage (possible in its own section or collapsible info blurb).